R/shiny-shims.R

```diff
@@ -0,0 +1,47 @@
+shinyInstalled <- local({
+  detected <- FALSE
+  function() {
+    if (detected) {
+      return(TRUE)
+    }
+
+    if (nzchar(system.file(package = "shiny"))) {
+      detected <<- TRUE
+      return(TRUE)
+    }
+
+    return(FALSE)
+  }
+})
+
+#' Get default reactive domain
+#'
+#' Pass-through to [shiny::getDefaultReactiveDomain()], unless the shiny package
+#' is not installed, in which case `NULL` is returned.
+#'
+#' @keywords internal
+#' @export
+getDefaultReactiveDomain <- function() {
```

Comment:
My worry is this name will cause confusion / namespace conflict. (Ex: `shiny::renderDataTable`)

`shinyGetDefaultReactiveDomain`?

Reply:
I don't suppose... I could make `shiny::getDefaultReactiveDomain()` the default value, but do some `missing()` stuff to not evaluate it if Shiny isn't installed??

Reply:
I could make the default `"shiny::getDefaultReactiveDomain()"` and hope nobody notices 😂

Reply:
Huh, the delayed evaluation approach works. Don't know if I love it, but it works.

Reply:
Like `getDefaultReactiveDomain <- shiny::getDefaultReactiveDomain` ? 🤣 No.  

Even when setting in inside `.onLoad`, there will be issues when people install shiny and haven't restarted yet

Reply:
Even still, I worry about the name clashing.